### ISSUE TITLE
GUI unresponsive on Godot 4 (regression from the NVIDIA Linux 450.66 driver and later)

### ISSUE THREAD

**Issue by Alexia-AT-Digitecnology** [godotengine/godot#41614]
***Bugsquad edit:** This issue has been confirmed several times already. No need to confirm it further.*

> **Workaround: Enable *Single Window Mode* in the Editor Settings, or use the `--single-window` command line argument and open a project directly by specifying the path to its `project.godot` file.**
>
> If you can't get to the checkbox, open `$HOME/.config/godot/editor_settings-4.tres` and add this line:
>
> ```tres
> interface/editor/single_window_mode = true
> ```

___

**Godot version:** v4.0 Git commit f10c3810bb0abfa18b1a579ee927f460bf191b6d

**OS:** Archlinux latest (NVIDIA GTX 1060, 64 GB RAM, Intel Core i7)

**Issue description:** When initialize godot UI modals remains unresponsive

**Steps to reproduce:**

- Compile godot
- Open godot
- Create a new project
- Specify directory
- UI remains unresponsive
![Godot bug UI unresponsive](https://user-images.githubusercontent.com/18708610/91649360-869a6900-ea49-11ea-849b-5097e4f5cacd.png)

*Bugsquad edit (keywords for easier searching): crash, crashing, freeze, freezing*
*Windows counterpart of this issue: https://github.com/godotengine/godot/issues/45325*

**Comment by Riteo** [comment]
Maybe related to #41574?

**Comment by Alexia-AT-Digitecnology** [comment]
Yeah, its related with that @Riteo. It's the same bug.

**Comment by akien-mga** [comment]
That's not i3wm like in #41574, is this KDE Plasma?

**Comment by akien-mga** [comment]
For the reference, I don't get this issue with a build from the current `master` branch on Mageia 8 with KDE Plasma 5 / KWin running on X11.

**Comment by Riteo** [comment]
@akien-mga what version of the NVidia driver do you have installed?

**Comment by akien-mga** [comment]
I'm #TeamAMD :)

Dual graphics Intel HD 630 / AMD Radeon RX Vega M both running on Mesa 20.1.6.

**Comment by Riteo** [comment]
@DavidNexuss could [replicate this issue on all the WMs he tried](https://github.com/godotengine/godot/issues/41574#issuecomment-682385003).
They also run Arch Linux with an NVIDIA card, so, I believe that this may be a duplicate of #41574.

**Comment by pouleyKetchoupp** [comment]
@Riteo I can't reproduce it with the recommended nvidia drivers (version 440) on Ubuntu 20.04.1 with Gnome or KDE (but I can reproduce it when using i3). I tried to install nvidia drivers version 450.66 but it caused all application windows to be either black or full of graphical glitches so it doesn't seem compatible with my configuration.

I'll try and see if I can install Arch Linux separately to reproduce the issue.

**Comment by seadra** [comment]
I can confirm this issue exists on Arch Linux + NVidia + KDE. Godot 4 is unusable due to this issue at the moment.

**Comment by Riteo** [comment]
@seadra did you start godot with only --single-window? IIRC after opening a project from the project manager, the editor is started without that flag.

**Comment by seadra** [comment]
Indeed! The problem disappears when editing a project directly with -e (and --single-window)!

This is indeed an issue related to multiple windows.

**Comment by pouleyKetchoupp** [comment]
I'd like to confirm this issue is similar to #41574. Could someone run these tests on ArchLinux with an nvidia card:

1. Simple vulkan test
-Install `vulkan-tools`
-Run `vkcube`
-Does it freeze when you focus in and out of the window?

2. Disable PRIME synchronization 
-Create file `/etc/modprobe.d/nvidia-graphics-drivers.conf`
-Add line `options nvidia-drm modeset=0`
-Run `sudo update-initramfs -u`
-Reboot
-Check if the problem persists in Godot (multi-window mode) and vkcube

**Comment by seadra** [comment]
I can confirm that `vkcube` indeed freezes momentarily, by simply moving the mouse in and out of the program window (not using PRIME, screen is directly hooked to NVIDIA card).

**Comment by pouleyKetchoupp** [comment]
Thanks @seadra, that confirms a general issue with the nvidia drivers and not just in godot. In your case it might work better by switching the drivers version to 450.57 (apart from the workaround with --single-window).

**Comment by Riteo** [comment]
@pouleyKetchoupp in my case after rolling back to 450.57

**Comment by Riteo** [comment]
@pouleyKetchoupp in my case, after rolling back to 450.57 vkcube still freezes briefly after focusing and unfocusing its window, but Godot 4.0 doesn't freeze the X11 server.

**Comment by pouleyKetchoupp** [comment]
@Riteo Could you please try and run `nvidia-bug-report.sh` with 450.66 drivers enabled and provide me with the generated `nvidia-bug-report.log.gz` file when you have some time?
Instructions are here: https://forums.developer.nvidia.com/t/if-you-have-a-problem-please-read-this-first/27131

I'm opening a ticket on the nvidia developer forum and it could help to provide more information. Unfortunately I still have issues with these drivers and they don't work at all for me :/

**Comment by pouleyKetchoupp** [comment]
I've opened the ticket upstream:
https://forums.developer.nvidia.com/t/x-server-freeze-when-creating-window-or-switching-focus-regression-from-450-66/155150

**Comment by Riteo** [comment]
@pouleyKetchoupp I too tought about making a ticket, but I thought that without a minimal Vulkan repo project we wouldn't get that far. I'll run that script as soon as I downgrade to 450.66! I tried to run the latest [455.23 beta](https://www.nvidia.com/en-us/drivers/results/163518/) and the issue persists, though there might be something that might be useful to you. Here's the changelog:
```
    Added support for the following GPUs:
        GeForce RTX 3080
        GeForce RTX 3090
        GeForce MX450
    Fixed a bug that prevented 8K displays from being used in portrait orientation on Pascal and earlier GPUs.
    Fixed a bug which caused excessive CPU usage in Vulkan applications which create a large number of VkFence objects. This was particularly prevalent in the Steam Play title Red Dead Redemption 2.
    Added support for a new device-local VkMemoryType which is host-coherent and host-visible. This may result in improved performance in some applications. Applications which are known to benefit from this new VkMemoryType include: DXVK Title DiRT Rally 2.0, Steam Play Title DOOM: Eternal, and DXVK title World of Warcraft.
    Fixed a bug that caused WebKit-based applications to crash when running on Wayland.
    Fixed a bug that led to display corruption at some resolutions when using an NVIDIA-driven display as a PRIME Display Offload sink.
    Fixed a bug in a SPIR-V optimization that may cause conditional blocks to not execute.
    Fixed a bug where calls to vkGetRandROutputDisplayEXT with unexpected inputs would generate X11 protocol errors. (https://bugs.winehq.org/show_bug.cgi?id=49407)
    Fixed a small memory leak during exit of the NVIDIA EGL driver.
    Implemented support for the NVIDIA NGX Updater. See the "NGX" chapter in the README for information on how to configure this feature.
    Removed support for the SLI modes "SFR", "AFR", and "AA".  SLI Mosaic, Base Mosaic, GL_NV_gpu_multicast, and GLX_NV_multigpu_context are still supported.
    Updated Base Mosaic support on GeForce to allow a maximum of five simultaneous displays, rather than three.
    Fixed several synchronization bugs that could momentarily lock up the X server when moving/resizing/focusing OpenGL and Vulkan windows when PRIME Sync was enabled.
    Added NVIDIA VDPAU driver support for decoding VP9 10- and 12-bit bitstreams. Note that VDPAU's presentation pipeline and OpenGL-VDPAU interop does not support 10- and 12-bit video surfaces yet. See the "VDPAU Support" appendix in the README for supported decoder profiles.
    Fixed a bug that could cause dual-link DVI to be driven over a connector that only supports single-link DVI, when "NoMaxPClkCheck" is specified in the "ModeValidation" X configuration option.  Note this fix may cause behavioral changes for configurations using this option.
    Fixed a regression that allowed displays to enter DPMS mode even when DPMS is disabled in the X server settings.
    Updated nvidia-settings to allow continued interaction with other pages and help content while editing application profiles.
```

**Comment by Riteo** [comment]
Ok, done.
What I did and what happened:
 - Start the X server with the flag `-logverbose 6`
 - Run Godot
 - Open a project with a window in it
 - The new window got created where i put my mouse. It didn't freeze, so I unfocused it and it froze for a few seconds.
 - Once unfrozen, waited a bit and it froze again for a few seconds.
 - Killed it from another terminal, froze again.
 - Dumped everything right away, here's the file: [nvidia-bug-report.log.gz](https://github.com/godotengine/godot/files/5272290/nvidia-bug-report.log.gz)

**Comment by seadra** [comment]
Issue persists in the current release 455.23.04

**Comment by pouleyKetchoupp** [comment]
Thanks both of you! I've updated the ticket upstream.

**Comment by aaronfranke** [comment]
I've been having the same problem with my GTX 1070, driver is `450.66-0ubuntu0.20.04.1`.

Workaround because nobody has mentioned it yet: Enable "Single window mode" and it seems to work fine.

**Comment by marcosbitetti** [comment]
the problem remains on Ubuntu 20.10, KDE environment and nVidia 455.38
vkcube workis fine

**Comment by auderer** [comment]
I'm also still experiencing this in 59b30e1d23d096f5faa36d2faaa07792eba4fd81:
Arch Linux, running nVidia 455.45.01 on GTX 1660 Ti

Single window mode resolved the issue for me.

**Comment by mrezai** [comment]
Hi
I have the same issue and did some tests.

Godot version: v4.0 Git commit 8a1c37dc
OS: Ubuntu 18.04.5
gcc: 7.5.0
GPU: Nvidia GT 1030
GPU driver: 450.80.02

Same issue happens when Godot compiled with `target=release_debug` and run in multiple windows mode but doesn't happen when Godot compiled with `target=debug`!
Since the unoptimized version is slower, I run release_debug version with valgrind to slow down it in runtime and as a result issue doesn't happen anymore!
From all of these can we conclude that at least in some situations, we get this issue because of the synchronization problem in Godot code itself?

Run Godot with valgrind(run too much slower, be patient!):
`valgrind bin/godot.linuxbsd.opt.tools.64 ~/dev/tests/godot/test-godot-v4-1/project.godot`

CPU/GPU time when Godot runs without valgrind:
![Screenshot from 2021-01-05 20-14-44](https://user-images.githubusercontent.com/5120436/103673559-fa69f680-4f92-11eb-9d04-82624245559d.png)

When runs with valgrind:
![Screenshot from 2021-01-05 18-37-42](https://user-images.githubusercontent.com/5120436/103673744-300edf80-4f93-11eb-893e-c46fd3b8ea4e.png)

**Comment by poiati** [comment]
It looks like Vulkan is in bad shape for Nvidia/linux users right now. I am using Ubuntu 20.04 and NVIDIA driver 455.38 and both vkcube and godot 4.0 runs very poorly. In Godot I believe I am also experiencing glitches since everything looks dark.

Have anyone tried NVIDIA beta drivers?

**Comment by Calinou** [comment]
> It looks like Vulkan is in bad shape for Nvidia/linux users right now. I am using Ubuntu 20.04 and NVIDIA driver 455.38 and both vkcube and godot 4.0 runs very poorly. In Godot I believe I am also experiencing glitches since everything looks dark.

Which graphics card model are you using? Here, Vulkan + NVIDIA looks fine to me in Godot and other games on a GTX 1080.

**Comment by poiati** [comment]
> > It looks like Vulkan is in bad shape for Nvidia/linux users right now. I am using Ubuntu 20.04 and NVIDIA driver 455.38 and both vkcube and godot 4.0 runs very poorly. In Godot I believe I am also experiencing glitches since everything looks dark.
> 
> Which graphics card model are you using? Here, Vulkan + NVIDIA looks fine to me in Godot and other games on a GTX 1080. Just make sure to [enable single-window mode](https://github.com/godotengine/godot/issues/41614#issue-688635392) to fix this regression in Godot.

![image](https://user-images.githubusercontent.com/21452/103927349-feb02400-50f8-11eb-91c1-37ebcf36a812.png)

I can run without crashing if I use single window mode but it is still not usable for other reason:

- Everytime I drag a window or change windows it runs very poorly, i can cleary see the trace of the window with a very low fps.
- There is this black background and every Mesh I add is 100% black before I adding light. Not sure if that is by design since in 3.X it does not work like this.

**Comment by Calinou** [comment]
> There is this black background and every Mesh I add is 100% black before I adding light. Not sure if that is by design since in 3.X it does not work like this.

The default environment hasn't been readded yet, that's why everything is black by default.

**Comment by poiati** [comment]
> > There is this black background and every Mesh I add is 100% black before I adding light. Not sure if that is by design since in 3.X it does not work like this.
> 
> The default environment hasn't been readded yet, that's why everything is black by default.

That makes sense. I can property see stuff after adding the environment and lighting. Any idea about what is going on with the FPS?

**Comment by Calinou** [comment]
> Any idea about what is going on with the FPS?

I can reproduce the issue when moving the window, but I don't think it's so critical for now. You generally don't move windows around all the time :slightly_smiling_face:

**Comment by Chaosus** [comment]
It sometimes happens on Windows 10 too.

**Comment by dsge** [comment]
This seems to still be an issue on the latest 460.23.03 nvidia driver on Ubuntu 20.04LTS. It still crashes without `interface/editor/single_window_mode = true`.

**Comment by Riteo** [comment]
@mrezai
> Since the unoptimized version is slower, I run release_debug version with valgrind to slow down it in runtime and as a result issue doesn't happen anymore!
From all of these can we conclude that at least in some situations, we get this issue because of the synchronization problem in Godot code itself?

Well, we already know that it's a synchronization issue of some sort, since the UI gets stuck on some Vulkan function binded to some related fences(see https://github.com/godotengine/godot/issues/41574#issuecomment-683549656).
It's interesting though that by having a slower framerate it doesn't freeze, and I can confirm that by putting a big enough delay right before all drawing code indeed avoids all the freezes. AFAICT since the editor doesn't do any frame limiting, it uses the GPU's frame synchronization fully, and for whatever reason(never used Vulkan, but to me it feels like the fence is never actually updated, and the X server eventually times out or something) the end of a frame draw is never signaled.

**Comment by MagdielM** [comment]
I am also having this issue on Windows 10 and on an AMD card. Not on startup, but upon switching back to Godot after focusing a different window.

**Comment by Riteo** [comment]
> I am also having this issue on Windows 10 and on an AMD card. Not on startup, but upon switching back to Godot after focusing a different window.

Mhhhh... Feels like a different issue, sort of. Can you send us a video?

**Comment by MagdielM** [comment]
> Mhhhh... Feels like a different issue, sort of. Can you send us a video?
Nevermind, turns out I just had a popup window minimized without noticing and it was stopping me from clicking anything in the main editor. Now that behavior's rather unintuitive, but I'm not sure if it's report-worthy.

**Comment by Riteo** [comment]
> Nevermind, turns out I just had a popup window minimized without noticing and it was stopping me from clicking anything in the main editor. Now that behavior's rather unintuitive, but I'm not sure if it's report-worthy.

No worries. I feel like this could confuse quite a few people though, so you should probably report that.

**Comment by underdoeg** [comment]
Can confirm the same issue on an up to date arch running gnome shell. 
Hardware is an RTX 2060 Super, with nvidia driver 460.39-1

**Comment by underdoeg** [comment]
Looks like the issue was fixed with the recent Nvidia 460.67 driver. I can open and edit the project settings in a new window.

**Comment by Riteo** [comment]
@underdoeg for me the issue still happens on a freshly updated arch linux install. It looked like it were a lot less "severe" in a way, almost as if it took more work on the GPU to get it to freeze though, but the X server still timeouts. I also read the last changelogs and didn't find anything about this issue and neither in the bug tickets, so I guess that either something changed, or it was all luck.

**Comment by underdoeg** [comment]
@Riteo yes, I was also surprised upon not finding anything in the changelog. But editor windows went from unusable to occasionally freezing on my arch machine with a Quadro RTX 4000. Last time I tried it was about a month ago. There was also an update to mutter in the meantime. if that is related at all.

**Comment by nathanfranke** [comment]
I can't reproduce this on Ubuntu 20.10, NVIDIA Driver 460.67, On my EVGA GTX 960 (Kernel 5.8.0-7642)

**Comment by TCROC** [comment]
I can confirm that I am also having this issue with Ubuntu 20.04, Gnome desktop, and and NVIDIA RTX 2070 Super graphics card with NVIDIA drivers.  If I select "Project", as soon as the popup appears (or very soon after) Godot freezes and becomes unresponsive.  I then have to kill it from the process manager.

**Comment by TCROC** [comment]
After discussing with @AndreaCatania in his Godex Discord server, we found a workaround.  This is what Andrea gave me and it works for me:

```
bin/godot --path path/to/project -e --single-window
```

**Comment by Riteo** [comment]
@TCROC you can also use the workaround written at the start of the issue description.

**Comment by TCROC** [comment]
@Riteo I just tested with the Godex build (which is based on Godot 4.0).  Yes it worked by first starting up with the terminal args ``bin/godot --path path/to/project -e --single-window`` then checking the checkbox.  Now I can double click on the binary as normal to start it up and open my projects.  Thanks!

The part that had me stuck was in every workaround, people said to use the ``--single-window`` flag, but didn't mention the need for the ``-e`` flag.  The ``-e`` flag is also required to start it up in ``single-window`` mode from the terminal.  At least that is what @AndreaCatania and I found when troubleshooting.

**Comment by Riteo** [comment]
@TCROC I should have pointed out that the workaround shown in the issue description only applies to the editor, not the project manager.

The -e flag is needed because after selecting a project from the project manager, a new process is started without the old arguments. Probably the best method would be starting the first time the project manager with `--single-window`, importing a project (process that requires multiple windows, hence the flag), editing the `editor_settings-4.tres` and then using godot normally.

**Comment by TCROC** [comment]
@Riteo Ah I see.  Should we update the workaround in the issue description for both cases?  I'm assuming that from my :+1: reactions, I'm not the only one that was confused by it.

**Comment by Riteo** [comment]
@TCROC I'm not sure. This issue actually started in #41574, where `--single-window` and it's repercussions in the project manager were discussed, but then branched here.

**Comment by NoFr1ends** [comment]
Driver version 465.27 seems too work more stable it still randomly freezes most of the time after having dropdowns visible, but not anymore all the time whenever opening a popup or similar.

**Comment by insomniacUNDERSCORElemon** [comment]
Not sure if it's relevant, but compiling with Clang* seems to eliminate (?) the random freezing from multi-windows. Still sluggish even in single-window mode, though.

*= Must add `use_static_cpp=no` to the command to get that to work (otherwise it gives the libatomic.a error)

**Comment by Riteo** [comment]
@insomniacUNDERSCORElemon mhh, that sounds weird, very weird. I'll investigate

**Comment by insomniacUNDERSCORElemon** [comment]
@Riteo I just found that I can easily re-create a crash in the project manager when trying to create a new project, but I still haven't seen a crash in the editor itself.

So there might be some difference there.

**Comment by Riteo** [comment]
@insomniacUNDERSCORElemon I still haven't found the occasion to test, but wait a second, you said that it's slugghish? Like, is the editor very slow? I think I know why it doesn't crash.

**Comment by insomniacUNDERSCORElemon** [comment]
@Riteo it was sluggish in the GCC version too (the one that did crash with sub-windows in the editor) so that's a shared thing with the Clang version. I mean it *could* be related, but I would assume it isn't.

**Comment by Riteo** [comment]
@insomniacUNDERSCORElemon maybe you're compiling it in debug mode? Anyways if it goes slow, it might not crash because the FPS are just low enough to not mess up the driver's synchronization (see [this comment](https://github.com/godotengine/godot/issues/41614#issuecomment-754760438) and [my response](https://github.com/godotengine/godot/issues/41614#issuecomment-757306127))

**Comment by insomniacUNDERSCORElemon** [comment]
@Riteo I didn't change the target. To be specific I am not aware of FPS, the sluggishness I'm experiencing is a delay when selecting nodes. Also seemingly any menu item (project or editor settings) that has color boxes in it. Which might even be the reason for the sluggishness w/node switching since they all have a color box (AFAICT) even if it's just modulation.

I couldn't find an FPS meter for the editor in 2D, trying to go to 3D immediately caused a freeze.

I'm now noticing freezing with the project/editor settings as well when typing something into the filter box. I've also seen 2 outright crashes due to the editor settings window. Even in the setting windows it's not a guaranteed thing and still less frequent than it was with GCC, so maybe it is related to FPS (the more complicated submenus not causing crashes)?

**Comment by Riteo** [comment]
@insomniacUNDERSCORElemon if you didn't specify the target then it chose the default, which is `debug`. That's probably why it's slow. Anyways, there's clearly something wrong with the drivers so I don't think that there's a way to get around it without fixing them, which we can't do now unless either nvidia magically released their source code or actually cared for their user bug reports a little more, which I don't think they never will. Our only hope at this point is reduz and his more direct contacts with various vendors including nvidia.

**Comment by insomniacUNDERSCORElemon** [comment]
Alright, I just tested it with the release_debug target and the result is that 1. Crashing is as frequent as it was with GCC debug 2. The sluggishness of switching nodes/colorboxes is gone/not noticable (even when crashing is present). 

Seems faster than GCC debug was, but maybe that's coincidence (newer master branch). Seems usable with multi-window-mode turned off.

In case anyone gets the idea, an external framelimit doesn't seem to prevent the crashing. At least for me, libstrangle seems to do nothing and mangohud seems to make stability significantly worse. I also tried disabling v-sync (for testing with 240 FPS on w/60Hz display) in addition to the framelimit.

**Comment by Calinou** [comment]
>  I also tried disabling v-sync (for testing with 240 FPS on w/60Hz display) in addition to the framelimit.

https://github.com/godotengine/godot/issues/36467

**Comment by carterisonline** [comment]
Thought I might as well chime in with what I found so far, since everything in this issue happened to me. This seems to be an issue for me with **everything** that uses vulkan on my system, including Retroarch, Dolphin Emulator, etc. @TCROC 's fix does work for me, so big thanks for that, but I've also noticed that this freezing only tends to happen when invoking certain animations; such as highlighting text, and scrolling with the scrollbar too fast.

My System: Linux 5.12.9-arch1-1, Arch Linux + KDE Plasma 5.21.5
GPU: NVidia GTX 1080 w/ NVidia 465.31-7 Drivers

**Comment by pouleyKetchoupp** [comment]
I've tried to reproduce this issue with no luck so far.
System: Arch Linux (kernel version 5.12.10-arch1-1)
NVidia card: GeForce 940M (drivers version 465.31)
Tested on Gnome 40.2.0 and KDE Plasma 5.22.1

@carterisonline Do you have a laptop optimus system or just a desktop Nvidia card? I had problems with Vulkan on my system (it was failing to run using the correct GPU) so I've followed this step:
https://wiki.archlinux.org/title/NVIDIA_Optimus#Use_NVIDIA_graphics_only
It's not ideal for battery usage, but things generally work much better with this configuration.

**Comment by Riteo** [comment]
@pouleyKetchoupp yep, I can still replicate this bug on 69553166ea. Had some big trouble doing it though. I had to compile it to `release_debug` and from the project manager I clicked on "About" and scrolled as fast as I could. Same old stacktrace coming from the drivers:
```
[1] /usr/lib/libc.so.6(+0x3cda0) [0x7f3859fbeda0] (??:0)
[2] /usr/lib/libGLX_nvidia.so.0(+0x4d686) [0x7f3858a36686] (??:0)
[3] /usr/lib/libnvidia-glcore.so.465.31(+0xe6c949) [0x7f38576f8949] (??:0)
[4] /usr/lib/libnvidia-glcore.so.465.31(+0x10e54fb) [0x7f38579714fb] (??:0)
[5] /usr/lib/libnvidia-glcore.so.465.31(+0x10e752e) [0x7f385797352e] (??:0)
[6] /usr/lib/libnvidia-glcore.so.465.31(+0x10de0c4) [0x7f385796a0c4] (??:0)
[7] /usr/lib/libnvidia-glcore.so.465.31(+0x11b65e3) [0x7f3857a425e3] (??:0)
[8] /usr/lib/libGLX_nvidia.so.0(+0x9359f) [0x7f3858a7c59f] (??:0)
[9] bin/godot.linuxbsd.opt.tools.64() [0x15cf79b] (/home/riteo/srg/godot/drivers/vulkan/vulkan_context.cpp:1927)
[10] bin/godot.linuxbsd.opt.tools.64() [0x158b93f] (/home/riteo/srg/godot/drivers/vulkan/rendering_device_vulkan.cpp:7669)
[11] bin/godot.linuxbsd.opt.tools.64() [0x304abd2] (/home/riteo/srg/godot/./core/templates/list.h:185)
[12] bin/godot.linuxbsd.opt.tools.64() [0x844f79] (/home/riteo/srg/godot/main/main.cpp:2511)
[13] bin/godot.linuxbsd.opt.tools.64() [0x81e419] (/home/riteo/srg/godot/platform/linuxbsd/os_linuxbsd.cpp:278)
[14] bin/godot.linuxbsd.opt.tools.64() [0x812682] (/home/riteo/srg/godot/platform/linuxbsd/godot_linuxbsd.cpp:58)
[15] /usr/lib/libc.so.6(__libc_start_main+0xd5) [0x7f3859fa9b25] (??:0)
[16] bin/godot.linuxbsd.opt.tools.64() [0x81cf0e] (??:?)
```

**Comment by pouleyKetchoupp** [comment]
@Riteo Thanks for info! Are you able to reproduce it consistently now or is it completely random? Also please share some details about your system and nvidia driver version.

**Comment by insomniacUNDERSCORElemon** [comment]
@pouleyKetchoupp I can consistently reproduce w/d36b220, GCC (Or Clang), target=release_debug, 5.12.9-arch1-1 (Arch Linux+XFCE+no compositor), 1050 Ti w/driver v 465.31. CPU is a Ryzen 2700, so there isn't a 2nd GPU for any confusion here.

The fastest way is to open any color picker and click+drag (although in debug_release+single window mode this seems to be broken anyways, as in dragging does not continue changing color. And GCC debug seems to have it working but not crashing from that. Crashing aside, different bug?).

A different way is by highlighting text in the search bar in the project or editor settings windows. I can also confirm it happens when scrolling the about page.

*In the past* I've also have had random freezing (and sometimes even crashing) with debug versions as I've commented here before. But as said in the past this is a very performance-dependent thing.

**Comment by john-sevsk** [comment]
For me It freezes only after moving cursor while holding mouse button inside text fields and scroll bars but not after selecting text only with keyboard or scrolling with mouse wheel.

Arch Linux 5.12.10-arch1-1
NVIDIA GeForce GTX 1070
NVIDIA Driver Version: 465.31
X.Org X Server 1.20.11
KDE Plasma: 5.22.1

**Comment by pouleyKetchoupp** [comment]
Thanks all for the replies and reproduction steps!

There are two last things that I would like to confirm before we can contact Nvidia support with more information:

1. Are you able to reproduce the issue with Calinou's nightly build?

Link:
https://archive.hugo.pro/builds/godot/master/editor/godot-linux-nightly-x86_64.zip

2. From what I understand, the easiest and most consistent reproduction steps are:
-Open any project or start a new one
-Open menu `Help` > `About Godot`
-Scroll up and down by clicking the slider with left mouse button and dragging until it freezes

Is that accurate? Are you always able to reproduce the issue on the first attempt, and if not how long does it take?

**Comment by insomniacUNDERSCORElemon** [comment]
@pouleyKetchoupp 

1. it's happening for me with that linked build. 
2. You can get it to happen even faster with the about menu in the project manager (bottom-right corner). Scrolling with the mousewheel works too, but seems to be linked to scrolling quickly (scrolling _slowly_ only seems to trigger a freeze w/dragging the mouse for whatever reason, mouse polling maybe? Similarly clicking on the scrollbar can cause a freeze if you use an autoclicker). Better via project manager as well, as it's multi-window by default unless you launch with a flag to disable that (not a project setting you can forget is the wrong value).

From what I'm seeing it is instant and consistent, at least triggered this way and from color/text boxes. I've tested it quite a few times, but I can't exactly test too much as it freezes my system for a second time when killing the process (xkill leaves them behind, taking up resources).

**Comment by Riteo** [comment]
Oh sorry @pouleyKetchoupp I completely forgot to answer! Like @insomniacUNDERSCORElemon I haven't got any iGPU since I run a ryzen 2600 and I can replicate this on the latest arch linux version too running version 465.31 of the drivers. Is that nightly build any special from a fresh compilation of HEAD? I can confirm though that it happens in a consistent way by doing things like scrolling or in general updating very quickly a sub-window. I don't know if the freeze depends on a continuous fast update or just a very well timed out one though. If you're contacting nvidia [I collected some detailed data using their debug tool some time ago](https://github.com/godotengine/godot/issues/41614#issuecomment-698005538), that may be useful to them. Also [I found out some time later that the X server timed out after the freeze](https://github.com/godotengine/godot/issues/41614#issuecomment-757306127) and I can confirm that it still happens, maybe they could use it as a starting point for finding out the cause of the issue.

**Comment by john-sevsk** [comment]
@pouleyKetchoupp
Yes. It's 100% reproducible with this build and these steps on the first attempt.
Btw I also have Ryzen 5 2600

**Comment by pouleyKetchoupp** [comment]
If you can reproduce this bug, please test this tentative fix from @reduz and let us know if it helps with the freeze when scrolling and the other random freezes related to NVidia GPU:
https://github.com/godotengine/godot/pull/49973
